python/ray/serve/api.py

```diff
@@ -588,8 +588,11 @@ def run(
             while True:
                 # Block, letting Ray print logs to the terminal.
                 time.sleep(10)
-        except KeyboardInterrupt:
+        except KeyboardInterrupt as e:
             logger.warning("Got KeyboardInterrupt, exiting...")
+            # We need to re-raise KeyboardInterrupt, so serve components can be shutdown
+            # from the main script.
+            raise e
```

Comment:
better to do just `raise` here to avoid polluting the stack trace (see https://stackoverflow.com/questions/36153805/difference-between-raise-and-raise-e)

Reply:
TIL 😄 Thanks Ed! let me push this change.